tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -525,43 +525,35 @@ void TTreeProcessorMT::Process(std::function<void(TTreeReader &)> func)
                                                           fFileNames, clusterAndEntries.second);
    }
 
-   const auto &clusters = clusterAndEntries.first;
-   const auto &entries = clusterAndEntries.second;
-
-   // Retrieve number of entries for each file for each friend tree
-   const auto friendEntries = hasFriends ? GetFriendEntries(fFriendInfo) : std::vector<std::vector<Long64_t>>{};
-
    // Parent task, spawns tasks that process each of the entry clusters for each input file
-   // TODO: for readability we should have two versions of this lambda, for shouldRetrieveAllClusters == true/false
-   auto processFile = [&](std::size_t fileIdx) {
-      // theseFiles contains either all files or just the single file to process
-      const auto &theseFiles = shouldRetrieveAllClusters ? fFileNames : std::vector<std::string>({fFileNames[fileIdx]});
-      // either all tree names or just the single tree to process
-      const auto &theseTrees = shouldRetrieveAllClusters ? fTreeNames : std::vector<std::string>({fTreeNames[fileIdx]});
-      // Evaluate clusters (with local entry numbers) and number of entries for this file, if needed
-      const auto theseClustersAndEntries =
-         shouldRetrieveAllClusters ? ClustersAndEntries{} : MakeClusters(theseTrees, theseFiles, maxTasksPerFile);
-
-      // All clusters for the file to process, either with global or local entry numbers
-      const auto &thisFileClusters = shouldRetrieveAllClusters ? clusters[fileIdx] : theseClustersAndEntries.first[0];
-
-      // Either all number of entries or just the ones for this file
-      const auto &theseEntries =
-         shouldRetrieveAllClusters ? entries : std::vector<Long64_t>({theseClustersAndEntries.second[0]});
-
+   auto processFileRetrievingAllClusters = [&](std::size_t fileIdx) {
       auto processCluster = [&](const EntryCluster &c) {
-         auto r = fTreeView->GetTreeReader(c.start, c.end, theseTrees, theseFiles, fFriendInfo, fEntryList,
-                                           theseEntries, friendEntries);
+         auto r = fTreeView->GetTreeReader(c.start, c.end, fTreeNames, fFileNames, fFriendInfo, fEntryList,
+                                           clusterAndEntries.second, GetFriendEntries(fFriendInfo));
          func(*r);
       };
+      fPool.Foreach(processCluster, clusterAndEntries.first[fileIdx]);
+   };
 
-      fPool.Foreach(processCluster, thisFileClusters);
+   auto processFileNotRetrievingAllClusters = [&](std::size_t fileIdx) {
+      // Evaluate clusters (with local entry numbers) and number of entries for this file
+      const auto theseClustersAndEntries = MakeClusters({fTreeNames[fileIdx]}, {fFileNames[fileIdx]}, maxTasksPerFile);
+      auto processCluster = [&](const EntryCluster &c) {
+         auto r =
+            fTreeView->GetTreeReader(c.start, c.end, {fTreeNames[fileIdx]}, {fFileNames[fileIdx]}, fFriendInfo,
```

Comment:
could put `{fTreeNames[fileIdx]}` and `{fFileNames[fileIdx]}` in dedicated variables for readability and to avoid repetition.

similarly for `theseClustersAndEntries.second[0]` and `theseClustersAndEntries.first[0]`